pallets/msa/src/lib.rs

```diff
@@ -112,13 +130,48 @@ pub mod pallet {
 		pub fn create(origin: OriginFor<T>) -> DispatchResult {
 			let who = ensure_signed(origin)?;
 
-			let (identifier, _) = Self::create_account(who.clone())?;
+			let (identifier, _) = Self::create_account(who.clone(), EMPTY_FUNCTION)?;
```

Comment:
I wish I could find a better alternative way instead of passing in an empty function, any alternative that would achieve the same effect is appreciated. (without any code duplication)

Reply:
Why not make the function deposit the event on success?

Reply:
sounds good